akka-http/src/main/resources/reference.conf

```diff
@@ -37,6 +37,13 @@ akka.http.routing {
   # for an entity data stream.
   decode-max-bytes-per-chunk = 1m
 
+  # Maximum content length after applying a decoding directive. When the directive
+  # decompresses, for example, an entity compressed with gzip, the resulting stream can be much
+  # bigger than the max-content-length. Like with max-content-length, this is not necessarilly a
+  # problem when consuming the entity in a streaming fashion, but does risk high memory use
+  # when the entity is made strict or marshalled into an in-memory object.
```

Comment:
Good explanation.

Reply:
Maybe add another comment, that this limit (like `max-content-length`) can be overridden on a case-by-case basis using `withSizeLimit`.